cmd/kube-controller-manager/app/options/concurrentresourcessyncs_options.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"github.com/spf13/pflag"
+	genericcontrollermanager "k8s.io/kubernetes/cmd/controller-manager/app"
+)
+
+// ConcurrentResourcesSyncsOptions is part of context object for the controller manager.
+type ConcurrentResourcesSyncsOptions struct {
+	ConcurrentJobSyncs           int32
+	ConcurrentDaemonSetSyncs     int32
+	ConcurrentEndpointSyncs      int32
+	ConcurrentRCSyncs            int32
+	ConcurrentRSSyncs            int32
+	ConcurrentDeploymentSyncs    int32
+	ConcurrentSATokenSyncs       int32
+	ConcurrentNamespaceSyncs     int32
+	ConcurrentResourceQuotaSyncs int32
+}
+
+// AddFlags adds flags related to debugging for controller manager to the specified FlagSet
+func (o *ConcurrentResourcesSyncsOptions) AddFlags(fs *pflag.FlagSet) {
+	if o == nil {
+		return
+	}
+
+	fs.Int32Var(&o.ConcurrentEndpointSyncs, "concurrent-endpoint-syncs", o.ConcurrentEndpointSyncs, "The number of endpoint syncing operations that will be done concurrently. Larger number = faster endpoint updating, but more CPU (and network) load")
```

Comment:
I tend to order them by controller, i.e. orthogonally to your order. @deads2k opinions?

Reply:
> I tend to order them by controller, i.e. orthogonally to your order. @deads2k opinions?

Yes, order by controller so we can start looking for commonality and split them out by controller too.

Reply:
done